code/lib/cli/src/automigrate/helpers/detectFramework.ts

```diff
@@ -0,0 +1,32 @@
+import type { PackageJson } from '@storybook/types';
+import { frameworkPackages } from '@storybook/core-common';
+import prompts from 'prompts';
+
+export const detectFramework = async (packageJson: PackageJson) => {
+  const allDependencies = {
+    ...packageJson.dependencies,
+    ...packageJson.devDependencies,
+    ...packageJson.peerDependencies,
+  };
+
+  // Pull the framework from dependencies in package.json
+  const matches = Object.keys(frameworkPackages).filter((pkg) => !!allDependencies[pkg]);
+  let [frameworkPackage] = matches;
+  if (matches.length > 1) {
```

Comment:
I know it's an edge case, but what happens if `matches` is 0. Do we want to let the user know, that his framework is not supported yet or we couldn't look up a framework?